### PR TITLE
List handling and variadic functions

### PR DESCRIPTION
**Pull request by prolic**
resolves https://github.com/prolic/fpp/issues/35

@joshhornby I added some more meaningful exceptions, can you check?

@paales This should fix your problems, can you verify and confirm?

@Maikel-Koek Wanna check this as well?

### CODE REVIEW

File: src/helpers.php [hunk 1]
```diff
@@ -216,30 +244,81 @@ function buildArgumentConstructorFromPayload(Argument $argument, Definition $def
...
         }
-    }
 
-    throw new \RuntimeException(sprintf(
-        'Cannot build argument constructor for %s',
-        $namespace !== '' ? $namespace . '\\' . $name : $name
-    ));
+        if (empty($method)) {
+            throw new \RuntimeException(sprintf(
+                'Cannot build argument constructor for %s, give a scalar type or a deriving like Enum, FromString, Uuid, FromScalar, FromArray',
```
Comment (prolic) [inline src/helpers.php]: @joshhornby see here f.e.

**Comment by prolic** [comment]
@Sevavietl wanna check this PR as well?

**Comment by Sevavietl** [comment]
@prolic, at first glance it looks great. It is holidays in my country, so I have not dived deep into the code. Will look closely later this week, if it still won't be merged. Thank you again for the great tool.

**Comment by prolic** [comment]
thanks @Sevavietl for reviewing

**Comment by prolic** [comment]
@paales can you also post definition of `\Bla\Project\Key` so I have the full picture?

**Comment by prolic** [comment]
@paales Or is is that `data ProjectKey = ProjectKey deriving (Uuid);` ?

**Comment by prolic** [comment]
@paales See https://github.com/prolic/fpp/pull/38/commits/6141b72953066b1abbe346675ffca65183fb0a20 - it allows commands and queries without any constructor arguments.

**Comment by paales** [comment]
@prolic It was `data Key = String deriving(FromString, ToString, Equals);`

**Comment by prolic** [comment]
@paales should work now. Can you confirm?

**Comment by paales** [comment]
@prolic Didn't have time to check yet, will do asap

**Pull request merged**